### ISSUE TITLE
[Bug]: @alpacahq/alpaca-trade-api uses outdated axios version 0.21.4

### ISSUE THREAD

**Issue by ManfredLange** [alpacahq/alpaca-trade-api-js#277]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

When I add @alpacahq/alpaca-trade-api version 3.1.2 (newest as of writing) as a dependency, this adds an indirect dependency of axios version 0.21.4 which contains a moderate vulnerability, see  https://github.com/advisories/GHSA-wf5p-g6vw-rhxx  for more details on that vulnerability.

### Expected Behavior

When I add `@alpacahq/alpaca-trade-api` it should add a or the most recent version of all its dependencies, e.g. axios version 1.7.8.

### SDK Version I encountered this issue in

alpaca-trade-api 3.1.2

### Steps To Reproduce

```markdown
1. In a node environment, add `@alpacahq/alpaca-trade-api` to the project
2. Use your favorite package manager to fix vulnerabilities, e.g. "pnpm audit fix" to check for vulnerabilities.
3. Use a tool like "pnpm why axios" to confirm that `@alpacahq/alpaca-trade-api` uses it as an indirect dependency
```

### Filled out the Steps to Reproduce section?

- [X] I have entered valid steps to reproduce my issue or have attached a minimally reproducible case in code that shows my issue happening; and understand that without this my issue will be flagged as invalid and closed after 30 days.

### Anything else?

It might make sense to consider a practice whereby keeping all dependencies on a recent version.

Also, it might make sense to consider making `@alpacahq/alpaca-trade-api` ESM-compatible. Based on the information I have at this time, `axios` is not ESM-compatible. A suitable replacement could be `node-fetch` which is ESM-compatible.

**Comment by chand1012** [comment]
Because Axios is such an old version, its missing an adapter for using fetch in environments where that's needed (for example, Cloudflare Workers). Here's the error I got when trying to use this module from within a worker. 
![CleanShot 2025-01-03 at 16 40 51](https://github.com/user-attachments/assets/1b0538ff-450b-4ef5-be6a-80c042b46b6c)

**Comment by ManfredLange** [comment]
Doesn't appear as if there is a lot of interest by Alpaca to keep these libraries up-to-date.

They should make a decision. Either look after these libraries - is it any better for other languages? - or archive the repo,

The worst thing is to just let it sit there in limbo. For people to have to find out by themselves is just a waste of time. 

We are considering to ditch this npm package and build our own with just what we need. That'd be a shame, though, as we wouldn't built the same functionality.

Another option would be to fork it and publish it as a "working" package. However, we don't work for free and Alpaca wants us to make use of their API-first brokerage and trading solution. If they want others to maintain this package, then they should indicate what they would be willing to contribute financially.

My two cents.

**Comment by chand1012** [comment]
@ManfredLange seems like the reason this library isn't actively developed is that they're working on an alternative library, [alpaca-ts](https://github.com/alpacahq/typescript-sdk). However I do agree that there should be some resemblance of support for this library, especially with all the open issues and PRs.

**Comment by vicary** [comment]
@ManfredLange Thinking Alpaca being an API-first broker, the current effort to maintain their SDK sends quite a different message TBH.

Quoting [my question on discard](https://discord.com/channels/878427978223194123/951500849077698560/1314479564293931048) regarding their TypeScript SDK,

![Image](https://github.com/user-attachments/assets/29d6b267-bbac-4f99-a56d-a96e8ef4f278)

It's disappointing especially when their TypeScript SDK looks much more polished and promising is also ditched, because the main author left their team.

**Comment by ManfredLange** [comment]
This issue is still present in version 3.1.3 of the package `@alpacahq/alpaca-trade-api`

**Comment by ManfredLange** [comment]
@vicary I totally agree. Claiming to be an API-first broker is one thing. Being one is a different one.

It's obviously a business decision whether to provide SDKs or libraries for any language. However, if they provide a library, then it should be properly maintained. If they don't want to properly maintain it, then they should say so. Everybody deserves clarity.

At I see it: At the moment, because of this uncertainty - are they or aren't they maintaining this npm package properly - there is no incentive in the community to create and maintain an alternative. Any such attempt would run the risk of being superseded the next moment, once someone at Alpaca gets the idea, "hey, let's do another release".

I am not expecting them to maintain or not maintain this package. That is their decision. I would very much appreciate, though, to hear an official statement from them one way or the other. With such a statement, we could then all move on and draw our own conclusions.

**Comment by Azein** [comment]
In the upcoming major release we decided to move forward with as few dependencies as possible, and maintain a steady security profile: https://github.com/alpacahq/alpaca-trade-api-js/blob/ts-alpha/package.json#L123-L126

Please check out the alpha channel release at https://www.npmjs.com/package/@alpacahq/alpaca-trade-api/v/4.0.0-alpha.0

@ManfredLange the official statement regarding the future of the SDK: https://github.com/alpacahq/alpaca-trade-api-js/discussions/296

The issue will remain open until the full release of 4.0

**Comment by Azein** [comment]
4.0 is live, closing the issue.

**Comment by ManfredLange** [comment]
Too late. I have implemented my own client using AI. If something changes, the AI agent just needs the OpenAPI specs plus some documentation. It will create tests, then creates or extends the API client class. Furthermore, my Alpaca client now has the minimal code for my use case.

**Comment by vicary** [comment]
@ManfredLange Ever thought about that they only start supporting this because they now have access to LLM?

LLM make implementations cheap, but relative effort and responsibility ownership shouldn't change.
